pandablocks/commands.py

```diff
@@ -229,49 +239,123 @@ def execute(self) -> ExchangeGenerator[None]:
         assert ex.line == "OK"
 
 
-class GetBlockNumbers(Command[Dict[str, int]]):
-    """Get the name and number of each block type in a dictionary,
-    alphabetically ordered
+@dataclass
+class GetBlockInfo(Command[Dict[str, BlockInfo]]):
+    """Get the name, number, and description of each block type
+    in a dictionary, alphabetically ordered
+
+    Args:
+        skip_description: If `True`, prevents retrieving the description
+            for each Block. This will reduce network calls.
 
     For example::
 
-        GetBlockNumbers() -> {"LUT": 8, "PCAP": 1, ...}
+         GetBlockInfo() ->
+             {
+                 "LUT": BlockInfo(number=8, description="Lookup table"),
+                 "PCAP": BlockInfo(number=1, description="Position capture control"),
+                 ...
+             }
     """
 
-    def execute(self) -> ExchangeGenerator[Dict[str, int]]:
+    skip_description: bool = False
+
+    def execute(self) -> ExchangeGenerator[Dict[str, BlockInfo]]:
         ex = Exchange("*BLOCKS?")
         yield ex
-        blocks_list = []
+
+        blocks_list, commands = [], []
         for line in ex.multiline:
             block, num = line.split()
             blocks_list.append((block, int(num)))
-        return dict(sorted(blocks_list))
+            commands.append(Get(f"*DESC.{block}"))
+
+        if self.skip_description:
+            description_values = tuple(None for _ in range(len(commands)))
```

Comment:
```suggestion
            description_values = [None for _ in commands]
```

Reply:
This fix fails mypy checks - as the other way we assign `description_values` is a `Tuple[Any, ...]`, we cannot use a list here. I also cannot just use `( ... )` to create the tuple, as MyPy says that's of type `"Generator[None, None, None]")`. So from what I can see the simplest solution is what was originally there.